.github/workflows/lint.yml

```diff
@@ -111,7 +111,7 @@ jobs:
     name: pr-sanity-checks
     runs-on: linux.20_04.4x
     # Only run this on pull requests
-    if: github.event_name == 'pull_request'
+    if: github.event_name == 'pull_request' && !contains(github.event.pull_request.labels.*.name, 'skip-pr-sanity-checks')
```

Comment:
I wonder if this would works. The list of labels comes from the pull request event, and AFAIK, it is when the PR is created.  Adding the label after that doesn't change anything.  To make this works, I remember you would need to push another change, i.e. rebasing, after attaching the label so that the latter event can pick it up.

Reply:
@huydhn You're correct (I tested it) - do you know of a way to get the preferred behavior where re-running could pick up the updated label even if nothing new has been pushed?

Reply:
There's a general solution using GitHub API that we are using here and there in the CI https://docs.github.com/en/rest/issues/labels.  Basically, this sample curl command will return the latest list of labels by the time of the call:

```
curl \
  -H "Accept: application/vnd.github+json" \ 
  -H "Authorization: token GITHUB_TOKEN" \
  https://api.github.com/repos/pytorch/pytorch/issues/PR_NUMBER/labels
```

I only wonder if running one additional call to GitHub API per workflow run is worth the effort for this rare case.  I assume that most people won't want to turn off this check.  May be it's easier to just have a higher limit for this check (> 2000). Just my two cents :)